### DISCUSSION TITLE
Sytem.IO.Ports is only allowed in windows

### DISCUSSION THREAD

**Discussion by thuhu** [microsoft/Dynamics365Commerce.Solutions#173] [Q&A] [answered]
We are developing a payment device extension for the Verifone device and ts connected to the POS windows machine using Serial Port. We can successfully communicate with the device from a net6 framework console app referencing the payment terminal project, however wehen a Store commerce app is created and installed it throw a runtime error "Sytem.IO.Ports is only allowed in windows".

**Comment by madyke** [comment, marked as answer]
Hello @thuhu - Do you know what version of System.IO.Ports you are referencing? It looks like the latest version should support .NET Standard 2.0+ and .NET Framework 4.6.2+, so I don't think we would expect any issues.

Also, are you trying to use this extension with Store Commerce for Windows? Or mobile devices as well? If you could provide a sample of your extension and its .csproj file that may help identify if something is misconfigured or missing.
